### DISCUSSION TITLE
How to disable the nerd font and Ligatures

### DISCUSSION THREAD

**Discussion by lotusirous** [ghostty-org/ghostty#3323] [Q&A] [answered]
I'm not happy for the default behavior that enables the nerd font and font Ligatures because it's ugly. Do we have any document to disable this behavior ? 

Thanks. 🙏

**Comment by AnirudhG07** [comment]
Hi!
check out https://ghostty.org/docs/config/reference
Anything you need to know related to changing fonts is present here!

Also you can check out the default configuration which contains hints regarding options for fonts.

**Comment by 00-kat** [comment]
More specifically, [`font-feature`](https://ghostty.org/docs/config/reference#font-feature):
> To disable programming ligatures, use -calt since this is the typical feature name for programming ligatures. To look into what font features your font has and what they do, use a font inspection tool such as [fontdrop.info](https://fontdrop.info/).

I'm not sure if it's possible to disable Nerd Font glyphs though, but if you don't use them you won't see them (disabling them would make them show up as boxes instead).

**Comment by rdlugosz** [comment, marked as answer]
@lotusirous Here's what you need to add to your config file to disable ligatures:
```
font-feature = -calt
font-feature = -liga
font-feature = -dlig
```

I think ligatures in code are [confusing](https://github.com/ghostty-org/ghostty/issues/3657) and [cause output issues](https://youtu.be/VUxMfyzTM_Y?si=6zHP6F_0iThRUnt0&t=1111), but others seem to like it.

Given [forty years](https://en.wikipedia.org/wiki/Xterm) of monospaced precedent I think on by default is an odd choice for a terminal, unless there's an easy & obvious way to disable it. Read the docs doesn't qualify as easy & obvious enough for a feature that's going to be toggled by... 30%? 50%? of users, especially when one of the [design guidelines](https://ghostty.org/docs/features#terminal-(vt)-features) is to 

> ...be the most compatible for legacy applications while also providing the most modern features for new applications.

Overall really impressed with Ghostty! Looking forward to using it as my daily driver

**Reply by lotusirous** [reply]
Thanks for your suggestion. I found the same config in the document. :D

**Reply by velicanu** [reply]
Thanks both for the question and for the answer. A challenge with telling new users to read the docs for is that many users won't even know what to search for. For example I had no idea that dots being different widths is called `ligatures` and wouldn't know to search or that would be what I would have to toggle from reading the documentation alone. Reading this thread on github is what taught me that, and many users would be similarly unaware of font technical terms.

**Reply by 00-kat** [reply]
> dots being different widths is called `ligatures`

Yeah, that's one of the more subtle ligatures commonly found in fonts; a few more obvious ones are `->`, `=>`, `>=` and `!=`.

**Comment by fzakaria** [comment]
This was informative.
I've been using bazel which uses `...` a lot and noticed it had odd spacing.
